iceoryx_posh/test/moduletests/test_popo_publisher_port.cpp

```diff
@@ -0,0 +1,390 @@
+    maybeCaproMessage = m_publisherPortRouDi.getCaProMessage();
+    EXPECT_TRUE(maybeCaproMessage.has_value());
+    auto caproMessage = maybeCaproMessage.value();
+    EXPECT_THAT(caproMessage.m_type, Eq(iox::capro::CaproMessageType::OFFER));
+    EXPECT_THAT(caproMessage.m_serviceDescription, Eq(iox::capro::ServiceDescription("a", "b", "c")));
+    EXPECT_THAT(caproMessage.m_subType, Eq(iox::capro::CaproMessageSubType::EVENT));
+    EXPECT_THAT(caproMessage.m_historyCapacity, Eq(0u));
+
+    // stop offering
+    m_publisherPortUser.stopOffer();
+    EXPECT_FALSE(m_publisherPortUser.isOffered());
+    maybeCaproMessage = m_publisherPortRouDi.getCaProMessage();
+    EXPECT_TRUE(maybeCaproMessage.has_value());
+    caproMessage = maybeCaproMessage.value();
+    EXPECT_THAT(caproMessage.m_type, Eq(iox::capro::CaproMessageType::STOP_OFFER));
+    EXPECT_THAT(caproMessage.m_serviceDescription, Eq(iox::capro::ServiceDescription("a", "b", "c")));
+}
+
+TEST_F(PublisherPort_test, offering_with_history)
+{
+    iox::popo::PublisherPortData m_publisherPortDataHistory{iox::capro::ServiceDescription("x", "y", "z"),
+                                                            "myApp",
+                                                            &m_memoryManager,
+                                                            iox::MAX_HISTORY_CAPACITY_OF_CHUNK_DISTRIBUTOR};
+    iox::popo::PublisherPortUser m_publisherPortUserHistory{&m_publisherPortDataHistory};
+    iox::popo::PublisherPortRouDi m_publisherPortRouDiHistory{&m_publisherPortDataHistory};
+
+    // offering
+    m_publisherPortUserHistory.offer();
+    EXPECT_TRUE(m_publisherPortUserHistory.isOffered());
+    auto maybeCaproMessage = m_publisherPortRouDiHistory.getCaProMessage();
+    EXPECT_TRUE(maybeCaproMessage.has_value());
+    auto caproMessage = maybeCaproMessage.value();
+    EXPECT_THAT(caproMessage.m_type, Eq(iox::capro::CaproMessageType::OFFER));
+    EXPECT_THAT(caproMessage.m_serviceDescription, Eq(iox::capro::ServiceDescription("x", "y", "z")));
+    EXPECT_THAT(caproMessage.m_subType, Eq(iox::capro::CaproMessageSubType::FIELD));
+    EXPECT_THAT(caproMessage.m_historyCapacity, Eq(iox::MAX_HISTORY_CAPACITY_OF_CHUNK_DISTRIBUTOR));
+}
+
+TEST_F(PublisherPort_test, allocate_free)
+{
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(0u));
+    auto maybeChunkHeader = m_publisherPortUser.allocateChunk(10u);
+    EXPECT_FALSE(maybeChunkHeader.has_error());
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1u));
+    auto chunkHeader = maybeChunkHeader.get_value();
+    m_publisherPortUser.freeChunk(chunkHeader);
+    // this one is not stored in the last chunk, so all chunks must be free again
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(0u));
+}
+
+
+TEST_F(PublisherPort_test, allocate_send_no_subscriber)
+{
+    auto maybeChunkHeader = m_publisherPortUser.allocateChunk(10u);
+    EXPECT_FALSE(maybeChunkHeader.has_error());
```

Comment:
All the ```EXPECT_**``` have been verified in the previous test and can be assumed working and therefore removed